corehq/apps/es/README.rst

```diff
@@ -80,6 +88,113 @@ This reindex procedure is inherently safe because:
   hoster are not outside the Dimagi main track.
 
 
+Design Details
+''''''''''''''
+
+Reindex Procedure Details
+'''''''''''''''''''''''''
+
+1. Configure multiplexing on an index.
+
+   - Configure the document adapter for the index with a "secondary index name".
+     This will cause the adapter to use multiplexing logic instead of a single
+     index.
+
+     **Note**: The multiplexing logic required for this operation is not yet
+     implemented. The multiplexing adapter will most likely delegate to two
+     document adapters configured for separate indexes. Suffice it to say that
+     when a secondary index is defined for an adapter, it effectively becomes a
+     multiplexing adapter (which to the consumer, is indistinguishable from a
+     "standard" adapter).
+
+   - *(Optional)* If the reindex involves other meta-index changes (shards,
+     mappings, etc), also update those configurations at this time.
+   - Add a migration which performs all cluster-level operations required for
+     the new (secondary) index. For example:
+
+     - creates the new index
+     - configures shards, replicas, etc for the index
+     - sets the index mapping
+
+   - Review, merge and deploy this change.  At Django startup, the new
+     (secondary) index will automatically and immediately begin receiving
+     document writes.
+
+2. Execute a management command to sync and verify the secondary index from the
+   primary.
+
+   **Note**: This command is not yet implemented.
+
+   This management command is idempotent and performs four operations in serial.
+   If any of the operations complete with unexpected results, the command will
+   abort with an error.
+
+   1. Executes a Elastic ``reindex`` request with parameters to populate the
+      secondary index from the primary, configured to not overwrite existing
+      documents in the target (secondary) index.
```

Comment:
> this will take data from the old index ... not ... from the primary data sources

Correct. This will be performed via Elasticsearch's Reindex API.